api/src/pcapi/core/subscription/api.py

```diff
@@ -435,21 +438,35 @@ def get_allowed_identity_check_methods(user: users_models.User) -> list[models.I
         if is_identity_check_with_document_method_allowed_for_underage(user):
             allowed_methods.append(
                 models.IdentityCheckMethod.UBBLE
-                if FeatureToggle.ENABLE_UBBLE.is_active()
+                if FeatureToggle.ENABLE_UBBLE.is_active() and has_allowed_ubble_if_subscription_overflow(user)
                 else models.IdentityCheckMethod.JOUVE
             )
 
     elif user.eligibility == users_models.EligibilityType.AGE18:
         if FeatureToggle.ALLOW_IDCHECK_REGISTRATION.is_active():
             allowed_methods.append(
                 models.IdentityCheckMethod.UBBLE
-                if FeatureToggle.ENABLE_UBBLE.is_active()
+                if FeatureToggle.ENABLE_UBBLE.is_active() and has_allowed_ubble_if_subscription_overflow(user)
                 else models.IdentityCheckMethod.JOUVE
             )
 
     return allowed_methods
 
 
+def has_allowed_ubble_if_subscription_overflow(user: users_models.User) -> bool:
+    if not FeatureToggle.ENABLE_UBBLE_LIMITATION_ON_SUBSCRIPTION_OVERFLOW.is_active():
+        return True
+
+    future_age = users_utils.get_age_at_date(
+        user.dateOfBirth,
+        datetime.datetime.utcnow() + datetime.timedelta(days=settings.UBBLE_SUBSCRIPTION_LIMITATION_DAYS),
+    )
+    if future_age > user.age and future_age in (15, 16, 17, 18):
```

Comment:
il faut rajouter 19 dans la liste, non ?

ou alors on veut pas laisser la priorité au pass-15-17 vs les 18 ?

si oui on peut utiliser `[age +1 for age in ELIGIBILITY_UNDERAGE_RANGE]` pour une meilleure lisibilité.